src/backend/src/router/triage_agent_router.py

```diff
@@ -38,58 +38,73 @@ def triage_agent_router(
-                                _logger.error(f"Raw JSON string: {last_text.text.value}")
-
-                                # If JSON parsing fails, handle retries or raise an error if max retries reached
-                                if attempt == max_retries:
-                                    raise RuntimeError(f"JSON parsing failed after {max_retries} attempts.")
-                                else:
-                                    # Exit the inner loop to retry agent run
-                                    _logger.info(f"Retrying agent run due to JSON parsing error... Attempt {attempt + 1}/{max_retries}")
-                                    break
+        try: 
+            # Process the agent run and handle retries
+            max_retries = int(os.environ.get("MAX_AGENT_RETRY", 3))
+            for attempt in range(1, max_retries + 1):
+                # Create thread for communication
+                thread = agents_client.threads.create()
+                _logger.info(f"Created thread, ID: {thread.id}")
+
+                # Create and add user message to thread
+                message = agents_client.messages.create(
+                    thread_id=thread.id,
+                    role="user",
+                    content=utterance,
+                )
+                _logger.info(f"Created message: {message['id']}")
+
+                # Create and process the agent run
+                run = agents_client.runs.create_and_process(thread_id=thread.id, agent_id=agent.id)
+                _logger.info(f"Run attempt {attempt} finished with status: {run.status}")
+
+                if run.status == "completed":
+                    # Fetch and log all messages if successful run
+                    messages = agents_client.messages.list(thread_id=thread.id, order=ListSortOrder.ASCENDING)
+                    for msg in messages:
+                        if msg.text_messages:
+                            last_text = msg.text_messages[-1]
+                            _logger.info(f"{msg.role}: {last_text.text.value}")
+
+                            # Load the agent response into a JSON
+                            if msg.role == "assistant" :
+                                # Check if agent response is able to be parsed
+                                try:
+                                    _logger.info(f"Agent response parsed successfully: {last_text.text.value}")
+                                    data = json.loads(last_text.text.value)
+                                    _logger.info(f"Agent response parsed successfully: {data}")
+                                    parsed_result = parse_response(data)
+                                    return parsed_result
+                                
+                                except Exception as e:
+                                    _logger.error(f"Agent response failed with error: {e}")
+                                    if attempt == max_retries:
```

Comment:
we have this same `attempt == ` logic in a couple places, can we try to refactor the code at all to minimize this?